doc/rtt.js

```diff
@@ -0,0 +1,178 @@
+
+// This is a mock module, exporting the function fingerprints that are defined somewhere
+// else in the .h / .cpp files.
+//
+// The goal is to be able to run JSdoc against this file to generate fancy API docs.
+//
+// TODO: Move these docstrings to the corresponding .h/.cpp file
+// TODO: Become able to run JSdoc in .h/.cpp files, with a JSdoc plugin a la https://www.npmjs.com/package/jsdoc-jsx
+
+/**
+ *
+ * The <tt>pc-nrfjprog-js.RTT</tt> module exposes the functionality to the RTT functionality.
+ *
+ * @example
+ * const nrfjprogjs = require('pc-nrfjprog-js');
+ * const RTT = nrfjprogjs.RTT;
+ *
+ * RTT.start(12345678, {}, function(err, down, up) {
+ *      RTT.write(0, 'Some data to write', function(err, length, timeSinceRTTStartInUs) {
+ *          RTT.read(0, 100, function(err, stringData, rawData, timeSinceRTTStartInUs) {
+ *              RTT.stop(function(err) {
+ *                  console.log('Stopped');
```

Comment:
Spaghetti code. Ugh. 😒 Hope we can switch to a `Promise` API in the future.

Reply:
I see your point.